File: Difficulty: Hard/Word Ladder I/word-ladder-i.cpp
class Solution {
  public:
    int wordLadderLength(string startword, string targetWord,
                         vector<string>& wordList) {
        // Code here
        queue<pair<string,int>> q;
        q.push({startword,1});
        unordered_set<string> st(wordList.begin(), wordList.end());
        st.erase(startword);
        
        while(!q.empty()){
            string word = q.front().first;
            int steps = q.front().second;
            q.pop();
            if(word==targetWord) return steps;
            for(int i=0;i<word.size();i++){
                char org = word[i];
                for(char j = 'a';j<='z';j++){
                    word[i] = j;
                    
                    if(st.find(word)!=st.end()){
                        st.erase(word);
                        q.push({word,steps+1});
                    }
                }
                word[i]=org;
            }

        }
        return 0;
    }
};

File: Difficulty: Hard/Word Ladder I/word-ladder-i.py
from collections import deque
class Solution:
	def wordLadderLength(self, startWord, targetWord, wordList):
		#Code here
		word_set = set(wordList)
        
        if targetWord not in word_set:
            return 0
            
        queue = deque([(startWord, 1)])
        
        while queue:
            word, steps = queue.popleft()
            
            if word == targetWord:
                return steps
                
            # Generate all possible transformations
            for i in range(len(word)):
                for ch in 'abcdefghijklmnopqrstuvwxyz':
                    if ch != word[i]:
                        new_word = word[:i] + ch + word[i+1:]
                        if new_word in word_set:
                            word_set.remove(new_word)
                            queue.append((new_word, steps + 1))
                            
        return 0
		    
		  
		